pyomo/core/expr/compare.py

```diff
@@ -196,7 +196,7 @@ def compare_expressions(expr1, expr2, include_named_exprs=True):
     )
     try:
         res = pn1 == pn2
-    except PyomoException:
+    except (PyomoException, AttributeError):
```

Comment:
Why do we now ignore AttributeErrors?

Reply:
The reverse polish notation uses tuples, and when a pyomo object is compared with a tuple using `==`, we get an `AttributeError`:

```
>>> import pyomo.environ as pe
>>> from pyomo.core.expr.compare import compare_expressions
>>> m = pe.ConcreteModel()
>>> m.x = pe.Var()
>>> m.y = pe.Var()
>>> e1 = m.x**(-1)*m.y
>>> e2 = m.y*(1/m.x)
>>> compare_expressions(e1, e2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../pyomo/core/expr/compare.py", line 198, in compare_expressions
    res = pn1 == pn2
          ^^^^^^^^^^
  File "/.../pyomo/core/expr/numeric_expr.py", line 421, in __eq__
    return _generate_relational_expression(_eq, self, other)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.../pyomo/core/expr/relational_expr.py", line 431, in _generate_relational_expression
    rhs = _process_relational_arg(rhs, 2)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.../pyomo/core/expr/relational_expr.py", line 413, in _process_relational_arg
    arg = _process_nonnumeric_arg(arg)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.../pyomo/core/expr/relational_expr.py", line 385, in _process_nonnumeric_arg
    if obj.is_component_type() and obj.is_indexed():
       ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'is_component_type'
```

Reply:
OK - this is really a bug in the relational expression generation system (which we want to revisit and move to using multiple dispatch)

Reply:
I'm not sure I follow. What is the expected behavior?